### PR TITLE
UM 10.8 updates

### PR DESCRIPTION
**Pull request by paulcresswell**
@dpmatthews Changes to the um-setup script that will be needed to run UM 10.8 on the VM, plus additional improvements for better error checking and to make it easier to use. Please review.

### CODE REVIEW

**Review by dpmatthews — APPROVED** [review]
Looks fine to me. Review requested from @scwhitehouse

File: usr/local/bin/um-setup [hunk 1]
```diff
@@ -2,45 +2,132 @@
...
-    exit 1
+function get_source_tree {
+  # Put a copy of the src $1 in tmpdir $2,
+  # whether a working copy or repository location,
+  # and return either the revision for a repos, or 'wc' for a working copy.
+  # This is because no check is made whether the wc contains local changes.
+  src=$1
+  tmpdir=$2
+
+  if [[ $src =~ ^fcm:|^svn:|file: ]]; then
```
Comment (scwhitehouse) [inline usr/local/bin/um-setup]: Do we have a requirement to support svn+ssh, http, etc?

File: usr/local/bin/um-setup [hunk 2]
```diff
@@ -76,23 +164,130 @@ rm -rf $tmpdir
...
+
+
+  # Remove any previous builds
+  make -f make/${vm_config}.mk clean > /dev/null 2>&1 
+
+  # Build the library
+  echo
+  echo "Building Shumlib $omp library..."
+  echo "Make output at $install_dir/make.log"
+  SHUM_OPENMP=$shum_omp make -f make/${vm_config}.mk shum_wgdos_packing &> "$install_dir/make.log"
```
Comment (scwhitehouse) [inline usr/local/bin/um-setup:267]: I'm probably missing something, but is this saying that if ``shum_omp=false`` then you want to run ``false make -f ....``? Should it be ``$omp make -f ..." instead? (I can't remember how that gets built...)

File: usr/local/bin/um-setup [hunk 3]
```diff
@@ -52,10 +139,11 @@ prefix="gcom_rosestem"
 mpichdir="vm_gfortran_mpich3/build"
 serialdir="vm_gfortran_serial/build"
 serial32dir="vm32B_gfortran_serial/build"
-if [ -d "$prefix$mpichdir" ] || [ -d "$prefix$serialdir" ] || [ -d "$prefix$serial32dir"]; then
+if [ -d "$prefix$mpichdir" ] || [ -d "$prefix$serialdir" ] || [ -d "$prefix$serial32dir" ]; then
   echo "Found one or more GCOM build directories to install"
 else
-  exit 1 "Cannot find a GCOM build in $cdir"
+  echo "Cannot find a GCOM build in $cdir"
```
Comment (scwhitehouse) [inline usr/local/bin/um-setup:205]: ``$gdir`` instead of ``$cdir``?

File: usr/local/bin/um-setup [hunk 4]
```diff
@@ -2,45 +2,132 @@
...
-    echo "$src is not a recognised repository location or a valid directory"
-    exit 1
+}
+
+function create_keyword {
+  # Create a (primary) keyword in keyword.cfg if it doesn't exist.
+  keyword=$1
+  keypath=$2
+  keyfile=$HOME/.metomi/fcm/keyword.cfg
+  if ! grep -qL "location{primary}\[$keyword\]" $keyfile; then
```
Comment (scwhitehouse) [inline usr/local/bin/um-setup]: This doesn't check that the keyword is set to the correct path, I think, just that it exists? (Unlike the rose.conf rose-ana database test later which does check that it's true).

**Review by scwhitehouse — CHANGES_REQUESTED** [review]
Nothing serious. One issue looks like a bug, one might be a bug but might also be me misunderstanding, and a couple of other things to consider.

I'm assuming this has been tested in on and off-line modes on the VM!

File: usr/local/bin/um-setup (hunk 1 above)
Comment (scwhitehouse) [reply usr/local/bin/um-setup]: I should add that "no" is perfectly acceptable as an answer from my point of view!

**Review by scwhitehouse — APPROVED** [review]
Those changes look fine to me!

File: usr/local/bin/um-setup (hunk 2 above)
Comment (scwhitehouse) [reply usr/local/bin/um-setup:267]: For future reference, I was missing something... this sets $SHUM_OPENMP=true (or false) before invoking make, and that's what Shumlib's make file requires...

File: usr/local/bin/um-setup (hunk 3 above)
Comment (scwhitehouse) [reply usr/local/bin/um-setup:205]: My mistake. Oops.

**Pull request merged**